Samples/AdapterSelection/AdapterSelection/cpp/main.cpp

```diff
@@ -124,6 +102,31 @@ int main(int argc, char* argv[])
 	PrintResults(resultVector);
 }
 
+LearningModelDevice getLearningModelDeviceFromAdapterIndex(int index, IDXGIFactory *pFactory) {
+
+    // create D3D12Device
+    com_ptr<IDXGIAdapter> spAdapter;
+    pFactory->EnumAdapters(index, spAdapter.put());
+    com_ptr<IUnknown> spIUnknownAdapter;
+    spAdapter->QueryInterface(IID_IUnknown, spIUnknownAdapter.put_void());
+    com_ptr<ID3D12Device> spD3D12Device;
+    D3D12CreateDevice(spIUnknownAdapter.get(), D3D_FEATURE_LEVEL_12_0, _uuidof(ID3D12Device), spD3D12Device.put_void());

```

Comment:
>D3D_FEATURE_LEVEL_12_0 [](start = 47, length = 22)

Curious : Why did you choose feature level 12?   Inside the winml runtime we create the device using feature level 11.

[https://docs.microsoft.com/en-us/windows/desktop/direct3d11/overviews-direct3d-11-devices-downlevel-intro](https://docs.microsoft.com/en-us/windows/desktop/direct3d11/overviews-direct3d-11-devices-downlevel-intro)

Reply:
hmm, I assumed that since its a d3d12 device that I would want feature level 12. This sample uses a D3D12 Command Queue to construct a LearningModelDevice, so I do need it to be a d3d12 device.

Reply:
nope, your good with feature level 11.   i would go with that, since it's the lowest common demoninator and gets us the most support for video cards.   feature level 11_0 is the very first feature level that d3d12 supported.   

[https://en.wikipedia.org/wiki/Feature_levels_in_Direct3D](https://en.wikipedia.org/wiki/Feature_levels_in_Direct3D) 

---
In reply to: [240838624](https://github.com/Microsoft/Windows-Machine-Learning/pull/107#discussion_r240838624) [](ancestors = 240838624)